### PR TITLE
Upgrade webpack and dependencies

### PR DESCRIPTION
**Pull request by TylerOSI**
## JIRA
No direct story

## Description
building the docker image was taking 10 minutes on every PR build. This should speed it up. old versions of webpacker use [extract-text-webpack-plugin](https://github.com/webpack-contrib/extract-text-webpack-plugin) which was slowing down compile time. Newer versions of webpacker use [mini-css-extract-plugin](https://github.com/webpack-contrib/mini-css-extract-plugin) instead.

... so i upgraded webpacker and a whole bunch of related dependencies. The application is running with styles in place, and tests seem to be passing so far.

## Tests

- [ ] I used TDD to develop the feature
- [ ] I have included unit tests
- [ ] I have included new acceptance tests or modified existing ones
- [ ] I have included other tests

## Types of changes

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] I ran all my tests locally which were green.
- [ ] My code adds no new issues to Code Climate
- [ ] I ran all the linters for the project.
- [ ] My code is in a stable state ready to be deployable, but not necessarily complete.
- [ ] I promise on my honor as a CWDS developer to ensure I don't break the build (or bring donuts if I leave things broken), to check linters, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

**Comment by TylerOSI** [comment]
Jenkins test this please

**Comment by TylerOSI** [comment]
just committed some changes to move around dependencies to keep the production build smaller

**Comment by LeonidMarushevskyi** [comment]
@TylerOSI Just FYI. This change makes no impact on IE11 performance.

**Comment by TylerOSI** [comment]
Jenkins please test this

File: docker/web/Dockerfile [hunk 1]
```diff
@@ -2,7 +2,7 @@ FROM cwds/cans_base_image:4
 WORKDIR /app
 ADD . /app
 RUN bundle install --jobs 20 --retry 5
-RUN SECRET_KEY_BASE=$(echo "basesecret$(echo $RANDOM)") NODE_ENV=production RAILS_ENV=production bundle exec rails assets:precompile
+RUN SECRET_KEY_BASE=$(echo "basesecret$(echo $RANDOM)") NODE_ENV=development RAILS_ENV=production bundle exec rails assets:precompile
```
Comment (denys-davydov) [inline docker/web/Dockerfile]: I have concerns with running node in development mode in our production docker image.

Comment (LeonidMarushevskyi) [reply docker/web/Dockerfile]: +1. Missed this.

Comment (TylerOSI) [reply docker/web/Dockerfile]: the alternative is putting a whole bunch of babel dependencies in development dependencies and bloating the production image. I get your concern about this though. maybe I should revert the last commit

**Comment by wmitchell-cambria** [comment]
@TylerOSI - Please run `yarn licenses-report` as part of this PR.

**Pull request merged**